src/registry.rs

```diff
@@ -93,11 +93,11 @@ impl RegistryCore {
 
     fn unregister(&mut self, c: Box<Collector>) -> Result<()> {
         let mut id_set = Vec::new();
-        let mut collector_id = 0;
+        let mut collector_id: u64 = 0;
         for desc in c.desc() {
             if id_set.iter().find(|id| **id == desc.id).is_none() {
                 id_set.push(desc.id);
-                collector_id += desc.id;
+                collector_id = collector_id.wrapping_add(desc.id);
```

Comment:
I think it is better to use `checked_add` and return an Error, rather than silently wrapping around.

Reply:
It is intended to overflow